src/main/java/com/videoplaza/dataflow/pubsub/source/task/convert/PubsubAttributeExtractor.java

```diff
@@ -0,0 +1,57 @@
+package com.videoplaza.dataflow.pubsub.source.task.convert;
+
+import com.google.protobuf.util.Timestamps;
+import com.google.pubsub.v1.PubsubMessage;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Map;
+import java.util.Optional;
+
+public class PubsubAttributeExtractor {
+
+   private static final Logger LOG = LoggerFactory.getLogger(PubsubAttributeExtractor.class);
+
+   private final String keyAttribute;
+   private final String timestampAttribute;
+
+   public PubsubAttributeExtractor(String keyAttribute, String timestampAttribute) {
+      this.keyAttribute = keyAttribute;
+      this.timestampAttribute = timestampAttribute;
+   }
+
+   public String getKey(Map<String, String> attributes, String defaultValue) {
+      return Optional
+          .ofNullable(getKey(attributes))
+          .orElse(defaultValue);
+   }
+
+   public String getKey(Map<String, String> attributes) {
+      return attributes.get(keyAttribute);
```

Comment:
would it make sense to return an optional here? given that it might return `null`

Reply:
All current external callers are happier with the null and need extra code to deal with optional. So null is fine here IMO